utils/find_missing_segbits.py

```diff
@@ -0,0 +1,221 @@
+                break
+
+        # Check if we should skip this tile
+        for pattern in args.skip_tiles:
+            if re.match(pattern, tile_type):
+                do_skip = True
+                break
+
+        if do_skip:
+            continue
+
+        logging.critical(tile_type)
+
+        # DB file names
+        tile_file = os.path.join(
+            args.db_root, "tile_type_{}.json".format(tile_type.upper()))
+        ppips_file = os.path.join(
+            args.db_root, "ppips_{}.db".format(tile_type.lower()))
+        segbits_file = os.path.join(
+            args.db_root, "segbits_{}.db".format(tile_type.lower()))
+
+        # Load pips
+        pips = read_pips_from_tile(tile_file)
+
+        # Load ppips (if any)
+        if os.path.isfile(ppips_file):
+            ppips = read_ppips(ppips_file)
+        else:
+            ppips = {}
+
+        # Load segbits (if any)
+        if os.path.isfile(segbits_file):
+            segbits = read_segbits(segbits_file)
+        else:
+            segbits = []
+
+        # There are non-pseudo pips in this tile
+        if len(pips):
+            missing_bits = 0
+            known_bits = 0
+
+            # Build a list of pips to check. If a pip is listed in the ppips
+            # file and it is not "default" then make it a pseudo one
+            pips_to_check = []
+            for pip in pips:
+                if pip in ppips.keys() and ppips[pip] != "default":
+                    continue
+                pips_to_check.append(pip)
+
+            # Missing segbits file
+            if len(segbits) == 0:
+                missing_bits = len(pips_to_check)
+                logging.critical(" MISSING: no segbits file!")
+
+            # Segbits file present
+            else:
+
+                # Check pips
+                for pip in pips_to_check:
+                    if pip not in segbits:
+
+                        # A "default" pip
+                        if pip in ppips.keys() and ppips[pip] == "default":
+                            missing_bits += 1
+                            logging.error(
+                                " WARNING: no bits for pip '{}' which defaults to VCC_WIRE"
+                                .format(pip))
+
+                        # A regular pip
+                        else:
+                            missing_bits += 1
+                            logging.error(
+                                " MISSING: no bits for pip '{}'".format(pip))
+
+                    # The pip has segbits
+                    else:
+                        known_bits += 1
+
+            # Report missing bit count
+            if missing_bits > 0:
+                logging.critical(
+                    " MISSING: no bits for {}/{} pips!".format(
+                        missing_bits, missing_bits + known_bits))
+            else:
+                logging.critical(" OK: no missing bits")
+
+        # No pips
+        else:
+            logging.warning(" OK: no pips")
```

Comment:
```suggestion
            logging.warning(" OK: no pips")
            return 0
```

Reply:
What is the difference ?

Reply:
Function returns with `0` and which is then propagated to `sys.exit(main(sys.argv))` below...

Reply:
Yep, that one I got. My question is about the other two issues.